File: api/submit.py
from fastapi import APIRouter, Request
from fastapi.responses import JSONResponse
import database
from models import RespuestaEncuesta, BARRIOS_GUAYAQUIL

router = APIRouter()


@router.post("/api/submit")
async def submit(respuesta: RespuestaEncuesta, request: Request):
    if respuesta.barrio not in BARRIOS_GUAYAQUIL:
        return JSONResponse(status_code=400, content={"ok": False, "error": "Barrio no válido"})

    if respuesta.email:
        respuesta.email = respuesta.email.strip().lower()

    ua = request.headers.get("user-agent", "")

    query = """
        INSERT INTO respuestas (
            email, tiene_app, es_emprendedor, profesion, edad_rango, barrio,
            p1, p2, p3, p4, p5, p6, p7, p7b, p8, p9,
            p10_mejoras, user_agent, duracion_segundos
        ) VALUES ($1,$2,$3,$4,$5,$6,$7,$8,$9,$10,$11,$12,$13,$14,$15,$16,$17,$18,$19)
        ON CONFLICT (email) DO UPDATE SET
            tiene_app = EXCLUDED.tiene_app,
            es_emprendedor = EXCLUDED.es_emprendedor,
            profesion = EXCLUDED.profesion,
            edad_rango = EXCLUDED.edad_rango,
            barrio = EXCLUDED.barrio,
            p1 = EXCLUDED.p1, p2 = EXCLUDED.p2, p3 = EXCLUDED.p3,
            p4 = EXCLUDED.p4, p5 = EXCLUDED.p5, p6 = EXCLUDED.p6,
            p7 = EXCLUDED.p7, p7b = EXCLUDED.p7b, p8 = EXCLUDED.p8, p9 = EXCLUDED.p9,
            p10_mejoras = EXCLUDED.p10_mejoras,
            created_at = NOW()
        RETURNING id
    """

    rid = await database.execute_insert(
        query,
        respuesta.email,
        respuesta.tiene_app,
        respuesta.es_emprendedor,
        respuesta.profesion,
        respuesta.edad_rango,
        respuesta.barrio,
        respuesta.p1, respuesta.p2, respuesta.p3,
        respuesta.p4, respuesta.p5, respuesta.p6, respuesta.p7, respuesta.p7b,
        respuesta.p8, respuesta.p9,
        respuesta.p10_mejoras[:1000],
        ua,
        respuesta.duracion_segundos,
    )

    return {"ok": True, "id": str(rid)}

File: submit.php
<?php
header('Content-Type: application/json; charset=utf-8');
require_once __DIR__ . '/config.php';
require_once __DIR__ . '/db.php';

if ($_SERVER['REQUEST_METHOD'] !== 'POST') {
    http_response_code(405);
    echo json_encode(["ok" => false, "error" => "Método no permitido"]);
    exit;
}

$input = json_decode(file_get_contents('php://input'), true);
if (!$input) {
    http_response_code(400);
    echo json_encode(["ok" => false, "error" => "JSON inválido"]);
    exit;
}

$email = strtolower(trim($input['email'] ?? ''));
if (!filter_var($email, FILTER_VALIDATE_EMAIL)) {
    http_response_code(400);
    echo json_encode(["ok" => false, "error" => "Email inválido"]);
    exit;
}

$existing = db_query_val("SELECT id FROM respuestas WHERE email = $1", [$email]);
if ($existing) {
    http_response_code(409);
    echo json_encode(["ok" => false, "error" => "Este email ya fue registrado. No se permiten respuestas duplicadas."]);
    exit;
}

$tiene_app = $input['tiene_app'] ?? '';
$es_emprendedor = $input['es_emprendedor'] ?? '';
if (!in_array($tiene_app, ['si', 'no']) || !in_array($es_emprendedor, ['si', 'no'])) {
    http_response_code(400);
    echo json_encode(["ok" => false, "error" => "Valores de filtrado inválidos"]);
    exit;
}

$profesion = $input['profesion'] ?? '';
$edad_rango = $input['edad_rango'] ?? '';
$barrio = $input['barrio'] ?? '';
if (!in_array($profesion, $PROFESIONES) || !in_array($edad_rango, $EDADES_RANGO) || !in_array($barrio, $BARRIOS_GUAYAQUIL)) {
    http_response_code(400);
    echo json_encode(["ok" => false, "error" => "Datos personales inválidos"]);
    exit;
}

$likert_fields = ['p1','p2','p3','p4','p5','p6','p7','p8','p9'];
foreach ($likert_fields as $f) {
    $val = intval($input[$f] ?? 0);
    if ($val < 1 || $val > 5) {
        http_response_code(400);
        echo json_encode(["ok" => false, "error" => "Valor inválido en $f"]);
        exit;
    }
}

$user_agent = $_SERVER['HTTP_USER_AGENT'] ?? '';
$p10_mejoras = substr(trim($input['p10_mejoras'] ?? ''), 0, 1000);
$duracion = intval($input['duracion_segundos'] ?? 0);

$query = "INSERT INTO respuestas (
    email, tiene_app, es_emprendedor, profesion, edad_rango, barrio,
    p1, p2, p3, p4, p5, p6, p7, p8, p9,
    p10_mejoras, user_agent, duracion_segundos, version_encuesta
) VALUES (
    $1, $2, $3, $4, $5, $6,
    $7, $8, $9, $10, $11, $12, $13, $14, $15,
    $16, $17, $18, $19
) RETURNING id";

$id = db_insert($query, [
    $email, $tiene_app, $es_emprendedor, $profesion, $edad_rango, $barrio,
    intval($input['p1']), intval($input['p2']), intval($input['p3']),
    intval($input['p4']), intval($input['p5']), intval($input['p6']), intval($input['p7']),
    intval($input['p8']), intval($input['p9']),
    $p10_mejoras, $user_agent, $duracion, '1.0'
]);

echo json_encode(["ok" => true, "id" => $id]);
